test/run_test.py

```diff
@@ -439,19 +449,35 @@ def run_test(
 
     # Can't call `python -m unittest test_*` here because it doesn't run code
     # in `if __name__ == '__main__': `. So call `python test_*.py` instead.
-    argv = [test_module + ".py"] + unittest_args
+    argv = [test_file + ".py"] + unittest_args
 
     os.makedirs(REPO_ROOT / "test" / "test-reports", exist_ok=True)
     log_fd, log_path = tempfile.mkstemp(
         dir=REPO_ROOT / "test" / "test-reports",
-        prefix="{}_".format(test_module.replace("\\", "-").replace("/", "-")),
+        prefix="{}_".format(test_file.replace("\\", "-").replace("/", "-")),
         suffix=".log",
     )
     os.close(log_fd)
     command = (launcher_cmd or []) + executable + argv
+    should_file_rerun = (
+        "--subprocess" not in command
+        and not RERUN_DISABLED_TESTS
+        and isinstance(test_module, ShardedTest)
+        and test_module.time
+    )
+    timeout = THRESHOLD * 2 if should_file_rerun else None
```

Comment:
Could you explain a bit more on how this timeout value is used for large test like test_ops?  My understand is that test_ops will be sharded using the `THRESHOLD = 60 * 10` from `test_selections`.  Each shard then should take less than THRESHOLD to run.  So you just have x2 the value as timeout here to not waiting forever for a hang shard. A comment here would be nice